pipeline/gnomad/download_gnomad_threepointone.py

```diff
@@ -0,0 +1,406 @@
+    region_query = """
+        {   region(chrom: "%s", start: %s, stop: %s, reference_genome: %s) {
+            variants(dataset: %s) {
+              variant_id
+            }
+          }
+        }
+    """
+    r_q = region_query % (chrom, start, stop, reference, dataset)
+    response = requests.post(
+        'http://gnomad.broadinstitute.org/api',
+        json={"query": r_q, "variables": {}},
+        headers={"content-type": "application/json"})
+    max_retries=5
+    retries = 0
+    while retries < max_retries:
+        try:
+            parse = json.loads(response.text)
+            variants = parse['data']['region']['variants']
+            if variants is not None:
+                return variants
+            else:
+                continue
+        except Exception as e:
+            logging.info(e)
+            print(f"coords to variants error {e}")
+            retries += 1
+            time.sleep(3)
+            continue
+    logging.info(f'Request for intronic variants non_cancer failed 5 attempts')
+
+
+def fetch_data_for_one_variant(variant_id, dataset, max_retries=5):
+    variant_query = """
+            {   variant(variantId: "%s", dataset: %s) {
+                variant_id
+                chrom
+                pos
+                ref
+                alt
+                genome {
+                  ac
+                  an
+                  af
+                  ac_hom
+                  faf95 {
+                    popmax
+                    popmax_population
+                  }
+                  filters
+                  populations {
+                    id
+                    ac
+                    an
+                    ac_hom
+                  }
+                }
+                flags
+                transcript_consequences {
+                  hgvsc
+                  lof
+                  lof_flags
+                  lof_filter
+                  major_consequence
+                  transcript_id
+                }
+              }
+            }
+      """
+    v_q = variant_query % (variant_id, dataset)
+    retries = 0
+    while retries < max_retries:
+        try:
+            # https://stackoverflow.com/questions/49064398/requests-exceptions-chunkedencodingerror-connection-broken-incompleteread0
+            response = requests.post(
+                'http://gnomad.broadinstitute.org/api',
+                json={ "query": v_q,
+                       "variables": {}},
+                headers={"content-type": "application/json"})
+            if response.ok:
+                time.sleep(0.1)
+                parse = json.loads(response.text)
+                print(f'success for variant {variant_id}')
+                return(parse['data']['variant'])
+            else:
+                print('rate limit reached, waiting 60 seconds...')
+                time.sleep(60)
+                continue
+        except json.decoder.JSONDecodeError:
```

Comment:
I wonder whether it is worth to retry in all exceptions cases. Also, you may want to take a closer look at the http error codes the API give. E.g. it may only be worth to retry in case of HTTP 429 ("Too Many Requests")